src/parser.cc

```diff
@@ -14,12 +14,25 @@
 
 #include "parser.h"
 
+#include <string.h>
+#include <exception>
+
 #include "disk_interface.h"
 #include "metrics.h"
+#include "thirdparty/inja/inja.hpp"
+#include "util.h"
 
 using namespace std;
 
-bool Parser::Load(const string& filename, string* err, Lexer* parent) {
+void registerCallbacks(inja::Environment& env) {
+  env.add_void_callback("warning", 1, [](inja::Arguments& args) {
```

Comment:
lambdas are not allowed until https://github.com/ninja-build/ninja/pull/2089 is merged